controller/linodemachine_controller_helpers.go

```diff
@@ -721,6 +721,27 @@ func createInstance(ctx context.Context, logger logr.Logger, machineScope *scope
 
 	machineScope.LinodeClient.OnAfterResponse(ctr.ApiResponseRatelimitCounter)
 	inst, err := machineScope.LinodeClient.CreateInstance(ctx, *createOpts)
+
+	// if instance already exists, we get 400 response. get respective instance and return
+	if linodego.ErrHasStatus(err, http.StatusBadRequest) {
```

Comment:
400 is a very generic error thrown by the linode API, could we check the error itself to confirm that an instance exists with the existing label before trying to fetch the instance to save some API calls?